sirepo/template/srw.py

```diff
@@ -1432,8 +1448,16 @@ def _enum_text(name, model, field):
     return ""
 
 
-def _export_rsopt_config(data):
-    return _write_rsopt_zip(data, _rsopt_jinja_context(data))
+def _export_rsopt_config(data, run_dir):
+    ctx = _rsopt_jinja_context(data)
+    f = _write_rsopt_zip(data, ctx)
+    if _SIM_DATA.is_for_ml(data.report):
+        s = f"{_SIM_DATA.EXPORT_RSOPT}_run.sh"
+        pkio.write_text(s, template_common.render_jinja(SIM_TYPE, ctx, s))
+        template_common.subprocess_output(
+            ["bash", s],
+        )
```

Comment:
You definitely don't want to run rsopt during write_parameters(). This is also why the background_percent_complete() isn't working. Instead, write out a parameters.py file which calls this. 

You can add "machineLearningAnimation" to the "not mpi" part of sim_data.srw.is_run_mpi() similar to the beamlineAnimation exception there. That way it won't run the parameters.py with mpi.

Reply:
[80c95c3](https://github.com/radiasoft/sirepo/pull/5216/commits/80c95c34ca203c39b621753679f9f84e0fb86eeb)
[85050ad](https://github.com/radiasoft/sirepo/pull/5216/commits/85050adcfd762b12f3dc087afbc7d7ce8e5476cd)
Running the bash script should not happen in mpi - rsopt takes care of it. If you try to run this parameters.py with mpi you get file descriptor errors